src/app/components/dungeon/dungeon.component.ts

```diff
@@ -13,21 +15,37 @@ export class DungeonComponent {
 
   private _dungeon: Dungeon;
   private _heroId: number;
+  private _hero: HeroModel;
   private _selectedObstacle: Obstacle;
+  private _events: any;
 
-  constructor(private dungeonService: DungeonService, private _route: ActivatedRoute) {
+  constructor(private dungeonService: DungeonService, private _route: ActivatedRoute, private _getSingleHero: HeroService) {
     this._heroId = Number(this._route.snapshot.paramMap.get('id'));
     this.dungeonService.getDungeon(this._heroId).subscribe(response => {
       this._dungeon = response;
       this._dungeon.classifyObstacles();
     });
+    this._getSingleHero.getHero(this._heroId)
+    .subscribe(data => this._hero = data);
   }
 
-  showSelectedItem(item) {
+  public showSelectedItem(item) {
     this._selectedObstacle = item;
   }
 
+  public onProgress(event) {
+    this._events = event [0];
+    this._hero = event [1];
+    const kiscica = new Dungeon(event[2]);
+    this._dungeon = kiscica;
```

Comment:
even if you need this variable for unknown reasons, name it better :)

Reply:
my fault lol

Reply:
solved